src/sentry/tasks/update_user_reports.py

```diff
@@ -28,16 +30,23 @@ def update_user_reports(**kwargs):
     updated_reports = 0
     samples = None
 
+    MAX_EVENTS = kwargs.get("max_events", 5000)
     for project_id, reports in project_map.items():
         event_ids = [r.event_id for r in reports]
         report_by_event = {r.event_id: r for r in reports}
-        snuba_filter = eventstore.Filter(
-            project_ids=[project_id],
-            event_ids=event_ids,
-            start=now - timedelta(days=2),
-            end=now + timedelta(minutes=5),  # Just to catch clock skew
-        )
-        events = eventstore.get_events(filter=snuba_filter)
+        # Batch these calls to avoid max query size errors
+        chunks = ceil(len(event_ids) / MAX_EVENTS)
```

Comment:
Nit: There's a helper for chunking in `sentry.utils.iterators.chunked`

Reply:
It's less memory efficient but I changed it.

Reply:
Yeah I noticed that too, we could potentially change the implementation of `chunked` to improve it